src/assets/scripts/commandParser/CommandParser.js

```diff
@@ -0,0 +1,305 @@
+     *
+     * @for CommandParser
+     * @method _buildCommandList
+     * @param commandArgSegments {array<string>}
+     * @return {array<CommandModel>}
+     * @private
+     */
+    _buildCommandList(commandArgSegments) {
+        let commandModel;
+
+        // TODO: this still feels icky and could be simplified some more
+        const commandList = _map(commandArgSegments, (commandOrArg) => {
+            if (commandOrArg === '') {
+                return;
+            } else if (REGEX.UNICODE.test(commandOrArg)) {
+                const commandString = unicodeToString(commandOrArg);
+                commandModel = new CommandModel(COMMAND_MAP[commandString]);
+
+                return commandModel;
+            } else if (_has(COMMAND_MAP, commandOrArg) && !this._isAliasCommandAnArg(commandModel, commandOrArg)) {
+                commandModel = new CommandModel(COMMAND_MAP[commandOrArg]);
+
+                return commandModel;
+            } else if (typeof commandModel === 'undefined') {
+                // if we've made it here and commandModel is still undefined, a command was not found
+                return;
+            }
+
+            commandModel.args.push(commandOrArg);
+        });
+
+
+        return _compact(commandList);
+    }
+
+    /**
+     * This method is used for addressing a very specific situation
+     *
+     * When the current command is `heading` and one of the arguments is `l`, the parser interprets
+     * the `l` as another command. `l` is an alias for the `land` command.
+     *
+     * This method expects that a commandString will look like:
+     * `AA321 t l 042`
+     *
+     * We look for the `heading` command and no existing arguments, as the `l` would become the
+     * first argument in this situation.
+     *
+     * @for CommandParser
+     * @method _isAliasCommandAnArg
+     * @param commandModel {CommandModel}
+     * @param commandOrArg {string}
+     * @return {boolean}
+     */
+    _isAliasCommandAnArg(commandModel, commandOrArg) {
+        if (!commandModel) {
+            return false;
+        }
+
+        return commandModel.name === 'heading' && commandModel.args.length === 0 && commandOrArg === 'l';
+    }
+
+    /**
+     * Fire off the `_validateCommandArguments` method and throws any errors returned
+     *
+     * @for CommandParser
+     * @method _validateAndParseCommandArguments
+     * @private
+     */
+    _validateAndParseCommandArguments() {
+        const validationErrors = this._validateCommandArguments();
+
+        if (validationErrors.length > 0) {
+            _forEach(validationErrors, (error) => {
+                throw error;
+            });
+        }
+    }
+
+    /**
+     * For each `CommandModel` in the `commandList`, first validate it's arguments
+     * then parse those arguments into a consumable array.
+     *
+     * @for CommandParser
+     * @method _validateCommandArguments
+     * @private
+     */
+    _validateCommandArguments() {
```

Comment:
Just one thing not working with the `hold` command still: the command alone with no arguments. This would be a standard holding pattern at the aircraft's present position (unrelated to a fix). All other combinations work properly.